### PR TITLE
Update eslint

### LINKED ISSUE #515: 'id-length' is invalid
Hi,

I cloned this repo and (installed the dependencies, and) ran `npm run lint`, and I'm getting this error:

> /Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config.js:141
>                     throw e;
>                     ^
>
> Error: eslint-config-airbnb:
> 	Configuration for rule "id-length" is invalid:
> 	Value "data["1"].properties" has additional properties.
> 
> Referenced from: /Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/.eslintrc
>     at validateRuleOptions (/Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config-validator.js:88:15)
>     at /Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config-validator.js:101:13
>     at Array.forEach (native)
>     at Object.validate (/Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config-validator.js: 100:35)
>     at loadConfig (/Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config.js:107:19)
>     at /Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config.js:135:46
>     at Array.reduceRight (native)
>     at loadConfig (/Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config.js:119:36)
>     at getLocalConfig (/Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config.js:243:23)
>     at Config.getConfig (/Users/Ari/Developer/airbnb-js-style/packages/eslint-config-airbnb/node_modules/eslint/lib/config.js:371:22)

I'm not sure what's causing it. I've just spend the past hour debugging it.

### PR DESCRIPTION
**Pull request by ariporad**
Update eslint to v1.4.x

Fixes #515.

### CODE REVIEW

File: packages/eslint-config-airbnb/test/test-base.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import test from 'tape';
 import base from '../base';
 
-test('base: does not reference react', t => {
-  t.plan(2);
+test('base: does not reference react', tst => {
```
Comment (ljharb) [inline packages/eslint-config-airbnb/test/test-base.js]: tbh this type of change to me is an indicator that perhaps we **don't** wan't `id-length` in the first place - `t` is the conventional argument name here for tape, and using an abbreviated "test" seems hacky.

Comment (justjake) [inline packages/eslint-config-airbnb/test/test-base.js]: yeah - but in that case we should ammend the styleguide as well to say 1-letter vars are ok in some cases

Comment (ljharb) [inline packages/eslint-config-airbnb/test/test-base.js]: I think they always make sense as iteration variables in loops, and in tape tests, to begin with.

**Comment by ljharb** [comment]
Thanks for your contribution! `eslint` has since been updated.

**Pull request closed without merging**